doc: can: fix build commands

nwsetec opened the pull request:
File prj.mcp2515.conf was previously removed from project as it was
made redundant.

Paths containing uppercase CAN were changed to lowercase.

martinjaeger commented on samples/drivers/can/README.rst:
```diff
@@ -43,7 +43,6 @@ provides the MCP2515 CAN controller:
    :zephyr-app: samples/drivers/CAN
```
`CAN` in this path should be lower-case. Maybe you can fix that aswell?

martinjaeger reviewed (commented):
Can you please

nwsetec committed:
doc: can: fix build commands

File prj.mcp2515.conf was previously removed from project as it was
made redundant.

Paths containing uppercase CAN were changed to lowercase.

Signed-off-by: Nick Ward <nick.ward@setec.com.au>
